lib/routes/devices-sessions.js

```diff
@@ -276,28 +276,20 @@ module.exports = (log, db, config, customs, push, devices) => {
         db.devices(uid)
           .then(deviceArray => {
             reply(deviceArray.map(device => {
-              if (! device.name) {
-                device.name = devices.synthesizeName(device)
-              }
-
-              if (! device.type) {
-                device.type = device.uaDeviceType || 'desktop'
+              return {
+                id: device.id,
+                isCurrentDevice: device.sessionToken === sessionToken.tokenId,
+                lastAccessTime: device.lastAccessTime,
+                lastAccessTimeFormatted: localizeTimestamp.format(
+                  device.lastAccessTime,
+                  request.headers['accept-language']
+                ),
+                name: device.name || devices.synthesizeName(device),
+                type: device.type || device.uaDeviceType || 'desktop',
+                pushCallback: device.pushCallback,
+                pushPublicKey: device.pushPublicKey,
+                pushAuthKey: device.pushAuthKey
               }
-
-              device.isCurrentDevice = device.sessionToken === sessionToken.tokenId
-
-              device.lastAccessTimeFormatted = localizeTimestamp.format(
-                device.lastAccessTime,
-                request.headers['accept-language']
-              )
-              delete device.sessionToken
-              delete device.uaBrowser
-              delete device.uaBrowserVersion
-              delete device.uaOS
-              delete device.uaOSVersion
-              delete device.uaDeviceType
```

Comment:
This approach of deleting properties we don't want and returning whatever's left is fundamentally error-prone. Far better to return a new object that only includes the stuff we actually want.